app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostFragment.kt

```diff
@@ -82,6 +82,7 @@ class ZimHostFragment : BaseFragment(), ZimHostCallbacks, ZimHostContract.View {
   private var ip: String? = null
   private lateinit var serviceConnection: ServiceConnection
   private var progressDialog: ProgressDialog? = null
+  private var booksList: List<BooksOnDiskListItem>? = null
```

Comment:
why null? an empty list will do for an initial state

Reply:
if we set empty list then it will not  show anything on adapter after this
```
setBooksAdapter(if (!ServerUtils.isServerStarted) hostBooks else booksList)
```

Reply:
we need to maintain a list of the currently available books and then at certain times show only a subset of these books and then other times revert to showing the full currently available. This PR seems a bit confused right now but stick to this flow. This screen could be greatly improved by a MVI rewrite but definitely another ticket.

Reply:
>This PR seems a bit confused right now but stick to this flow.

@macgills what should we do now?

Reply:
Follow the flow.
AllBooks starts empty
When the books are loaded set AllBooks to the books
submit the items to the list
decide on external state to show everything or only selection
when this external state changes then show everything again

I am just overall confused about `booksList` and when/why/where you are using it